### PR TITLE
ci: adapt release for `cargo-binstall`

### PR DESCRIPTION
**Pull request by Coekjan**
Adapt the release artifact names, so that `cargo-binstall` can find `typstyle-{ target }` in github release assets.

Notes:
- `cargo-binstall` sets file-mode for binaries ( https://github.com/cargo-bins/cargo-binstall/blob/76814e4e8feaede87d8d112a4fad519d8eeaf81d/crates/binstalk-bins/src/lib.rs#L218 ), so current permission-loss artifacts simply work well.

### CODE REVIEW

**Comment by Enter-tainer** [comment]
cc our aur package maintainer @jisu-woniu . You may need change typstyle-bin's packaging script in next release

File: .github/workflows/release.yml
Comment (Enter-tainer) [inline .github/workflows/release.yml:133]: what is rust-target? is it defined by `dtolnay/rust-toolchain` ?

**Review by Enter-tainer — APPROVED** [review]
cool

**Comment by Coekjan** [comment]
something bad. let me fix it now.

File: .github/workflows/release.yml
Comment (Coekjan) [reply .github/workflows/release.yml:133]: `rust-target` is defined above (`strategy.matrix`)

**Comment by Coekjan** [comment]
Please reapprove the workflow, and we shall check the artifacts carefully before merging.

**Comment by Coekjan** [comment]
Artifacts look good.

**Review by Enter-tainer — APPROVED** [review]

**Pull request merged**
